online_meetings/canvas_zoom_meetings.py

```diff
@@ -0,0 +1,139 @@
+                # Get the form and parse out all of the inputs
+                form = soup.find('form')
+                if not form:
+                    logger.info("Could not find a form to launch this zoom page, skipping")
+                    break
+
+                fields = form.findAll('input')
+                formdata = dict((field.get('name'), field.get('value')) for field in fields)
+
+                # Get the URL to post back to
+                posturl = form.get('action')
+
+                # Start up the zoom session
+                zoom_s = requests.Session()
+                # Initiate the LTI launch to Zoom in a session
+                r = zoom_s.post(url=posturl, data=formdata)
+
+                # Get the XSRF Token
+                pattern = re.search('"X-XSRF-TOKEN".* value:"(.*)"', r.text)
+
+                zoom_courses.append({
+                    'account_id': course.account_id, 'course_id': course.id, 'course_name': course.name
+                })
+
+                if pattern:
+                    zoom_s.headers.update({
+                        'X-XSRF-TOKEN': pattern.group(1)
+                    })
+                # Get tab 1 (Previous Meetings)
+                    data = {'page': 1,
+                            'total': 0,
+                            'storage_timezone': 'America/Montreal',
+                            'client_timezone': 'America/Detroit'}
+                    r = zoom_s.get("https://applications.zoom.us/api/v1/lti/rich/meeting/history/COURSE/all", params=data)
+                    zoom_json = json.loads(r.text)
+
+                    for meeting in zoom_json["result"]["list"]:
+                        zoom_courses_meetings.append({
+                            'course_id': course.id,
+                            'meeting_id': meeting['meetingId'],
+                            'meeting_number': meeting['meetingNumber'],
+                            'host_id': meeting['hostId'],
+                            'topic': meeting['topic'],
+                            'join_url': meeting['joinUrl'],
+                            'start_time': meeting['startTime'],
+                            'status': meeting['status'],
+                            'timezone': meeting['timezone']
+                        })
+    return (zoom_courses, zoom_courses_meetings)
+
+
+start_time = datetime.now()
+logger.info(f"Script started at {start_time}")
+
+(zoom_courses, zoom_courses_meetings) = zoom_course_report(ENV.get("CANVAS_ACCOUNT_ID", 1), ENV.get("CANVAS_TERM_ID", 1), True)
+
+zoom_courses_df = pd.DataFrame(zoom_courses)
+zoom_courses_df.index.name = "id"
+zoom_courses_meetings_df = pd.DataFrame(zoom_courses_meetings)
+zoom_courses_meetings_df.index.name = "id"
+
+zoom_courses_df.to_csv("zoom_courses.csv")
+zoom_courses_meetings_df.to_csv("zoom_courses_meetings.csv")
+
+end_time = datetime.now()
+logger.info(f"Script finished at {start_time}")
+print('Duration: {}'.format(end_time - start_time))
```

Comment:
Should also probably be a logging statement.

Reply:
Yup, thanks